source/iterator/iterator.go

```diff
@@ -110,29 +100,36 @@ func New(
 	}, nil
 }
 
-func setupSnapshotIterator(
-	ctx context.Context,
-	snowflake Repository,
-	table, key string,
-	columns []string,
-	element, offset, batchSize int,
-	isFirstStart bool,
-) (*SnapshotIterator, error) {
-	var (
-		index int
-	)
+func prepareCDC(ctx context.Context, snowflake *repository.Snowflake, table string) error {
+	// Check if table tracking table exist.
+	isTableExist, err := snowflake.IsTableExist(ctx, getTrackingTable(table))
+	if err != nil {
+		return fmt.Errorf("check if table exist: %w", err)
+	}
 
-	if !isFirstStart {
-		index = element + 1
+	if !isTableExist {
+		// Prepare tracking table for consume stream.
+		err = snowflake.CreateTrackingTable(ctx, getTrackingTable(table), table)
+		if err != nil {
+			return fmt.Errorf("create tracking table: %w", err)
+		}
 	}
```

Comment:
Is it possible to write "create if not exists"?

Reply:
let me recheck docs

Reply:
Yeah, but as for me better to check if table exist, because we add to this table additional fields and when  we start connectors and use query ```create table if exist```  we can't know if table  is exist and we don't do we need add new fields (it is can be case when created snowflake connector (tracking table was created), removed it and created one more time the same source connector)